source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -23,34 +23,73 @@ namespace PlayFab
         maximalNumberOfBatchesInFlight(defaultMaxBatchesInFlight),
         readBufferWaitTime(defaultReadBufferWaitTimeInMs),
         authenticationContext(nullptr),
-        emitType(PlayFabEventPipelineType::PlayFabPlayStream)
+        emitType(PlayFabEventPipelineType::PlayFabPlayStream),
+        useBackgroundThread(true)
     {
     }
 
-    PlayFabEventPipelineSettings::PlayFabEventPipelineSettings(PlayFabEventPipelineType type) :
+    PlayFabEventPipelineSettings::PlayFabEventPipelineSettings(PlayFabEventPipelineType type, bool useBackgroundThread) :
         bufferSize(defaultBufferSize),
         maximalNumberOfItemsInBatch(defaultMaxItemsInBatch),
         maximalBatchWaitTime(defaultMaxBatchWaitTimeInSeconds),
         maximalNumberOfBatchesInFlight(defaultMaxBatchesInFlight),
         readBufferWaitTime(defaultReadBufferWaitTimeInMs),
         authenticationContext(nullptr),
-        emitType(type)
+        emitType(type),
+        useBackgroundThread(useBackgroundThread)
     {
     }
 
     PlayFabEventPipeline::PlayFabEventPipeline(const std::shared_ptr<PlayFabEventPipelineSettings>& settings) :
+        batchCounter(0),
         buffer(settings->bufferSize),
         isWorkerThreadRunning(false)
     {
         eventsApi = std::make_shared<PlayFabEventsInstanceAPI>(PlayFabSettings::staticPlayer);
 
         this->settings = settings;
         this->batchesInFlight.reserve(this->settings->maximalNumberOfBatchesInFlight);
-        this->Start();
+        if (this->settings->useBackgroundThread)
+        {
+            this->Start();
+        }
     }
 
     PlayFabEventPipeline::~PlayFabEventPipeline()
     {
+        Stop();
+    }
+
+    void PlayFabEventPipeline::Start()
+    {
+        if (!this->settings->useBackgroundThread)
+        {
+            LOG_PIPELINE("PlayFabEventPipeline is set to NOT use background threads. Start() is not needed to be called then, but Update() is required to be manually called every tick in this case.");
+            return;
+        }
+
+        if(isWorkerThreadRunning)
+        {
+            LOG_PIPELINE("PlayFabEventPipeline has already been started, and should not be started again until Stop() is called.");
+            return;
+        }
+
+        // start worker thread
+        this->isWorkerThreadRunning = true;
+        if (!this->workerThread.joinable())
+        {
+            this->workerThread = std::thread(&PlayFabEventPipeline::WorkerThread, this);
+        }
+    }
+
+    void PlayFabEventPipeline::Stop()
+    {
+        if(!isWorkerThreadRunning)
```

Comment:
should there be a similar if check for if the settings don't specify `useBackgroundThread`?